.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,33 @@
+name: Benchmark
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+permissions:
+  contents: write
```

Comment:
Allowing write permission to the repository contents for this action is frightening. It enables rewriting the code base and is a prime target for an attack.

I do not feel the benefit out weigh the risks here. At the very least I think we need to carefully consider this and make sure we are making an informed decision.

Reply:
With protected branches, I think the token still couldn't write to the `main` branch, only to other ones.